### PR TITLE
Mat 7319 definition expression editor

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-7319](https://jira.cms.gov/browse/MAT-7319)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is in to the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/CqlBuilderPanel/definitionsSection/expressionSection/ExpressionEditor.tsx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                    if (val) {
+                      const type = availableTypes.find((el) => el === val);
+                      setNamesOptions(getNameOptionsByType(type));
+                      return type;
+                    }
+                    return val;
+                  }}
+                  options={renderMenuItems(availableTypes)}
+                  disabled={!canEdit}
+                  {...formik.getFieldProps("type")}
```
Comment (adongare) [inline src/CqlBuilderPanel/definitionsSection/expressionSection/ExpressionEditor.tsx]: I think ideally this should come first before all other overridden props

File: src/CqlBuilderPanel/definitionsSection/expressionSection/ExpressionEditor.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      <ExpandingSection
+        title="Expression Editor"
+        showHeaderContent={expressionEditorOpen}
+        children={
+          <>
+            <div tw="flex flex-wrap">
+              <div tw="w-1/2">
+                <Select
+                  label="Type"
+                  id={"type-selector"}
```
Comment (adongare) [inline src/CqlBuilderPanel/definitionsSection/expressionSection/ExpressionEditor.tsx]:
```suggestion
                  id="type-selector"
```

File: src/CqlBuilderPanel/definitionsSection/expressionSection/ExpressionEditor.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                  inputProps={{
+                    "data-testid": "type-selector-input",
+                  }}
+                  data-testid={"type-selector"}
```
Comment (adongare) [inline src/CqlBuilderPanel/definitionsSection/expressionSection/ExpressionEditor.tsx]:
```suggestion
                  data-testid="type-selector"
```

File: src/CqlBuilderPanel/definitionsSection/expressionSection/ExpressionEditor.tsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+                  }}
+                  options={renderMenuItems(availableTypes)}
+                  disabled={!canEdit}
+                  {...formik.getFieldProps("type")}
+                />
+              </div>
+              <div tw="flex-grow pl-5">
+                <AutoComplete
+                  label="Name"
+                  id={"name-selector"}
```
Comment (adongare) [inline src/CqlBuilderPanel/definitionsSection/expressionSection/ExpressionEditor.tsx]:
```suggestion
                  id="name-selector"
```

File: src/CqlBuilderPanel/definitionsSection/expressionSection/ExpressionEditor.tsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+                data-testid="expression-insert-btn"
+                disabled={
+                  !canEdit || !formik.values.type || !formik.values.name
+                }
+              >
+                Insert
+              </Button>
+            </div>
+            <div style={{ marginBottom: "72px" }} />
+            <div className={"full-row"}>
```
Comment (adongare) [inline src/CqlBuilderPanel/definitionsSection/expressionSection/ExpressionEditor.tsx]:
```suggestion
            <div className="full-row">
```

**Review by adongare — DISMISSED** [review]

File: src/CqlBuilderPanel/definitionsSection/expressionSection/ExpressionEditor.tsx (hunk 1 above)
Comment (sb-cecilialiu) [reply src/CqlBuilderPanel/definitionsSection/expressionSection/ExpressionEditor.tsx]: @adongare I just made the change, can you please check again? Thanks so much!!

**Review by adongare — APPROVED** [review]
